tensorflow/core/profiler/internal/tfprof_code.cc

```diff
@@ -421,14 +421,15 @@ void TFCode::AddNode(TFGraphNode* node) {
   // TODO(xpan): Consider to release CodeDef after TFCode is built. It
   // takes a lot of memory.
   std::set<string> traces;
-  for (int i = 0; i < node->call_stack()->traces().size(); ++i) {
+  for (int i = 0, end = node->call_stack()->traces().size(); i < end; ++i) {
     // Unlike op name, which is globally unique, trace name is only unique
     // w.r.t. it's parent.
     const string& trace = GetTraceString(node->call_stack()->traces().at(i));
     traces.insert(trace);
     pre_code_node = pre_code_node->AddChildren(
         trace, &node->call_stack()->traces().at(i), "");
-    if (i == node->call_stack()->traces().size() - 1) {
+    const int64 node_call_stack_traces_size_minus_one = node->call_stack()->traces().size() - 1;
+    if (i == node_call_stack_traces_size_minus_one) {
```

Comment:
`last_index` is much more readable. Consider what happens on a refactor where `call_stack()` gets changed to return an object which now has a `num_traces()` method so you don't need to do `->traces().size()` (a change which can be done automatically by tooling). Then the proposed name will no longer be valid.

Reply:
updated.